azimuth/config.py

```diff
@@ -358,6 +358,15 @@ def check_pipeline_names(cls, pipeline_definitions):
             raise ValueError(f"Duplicated pipeline names {pipeline_names}.")
         return pipeline_definitions
 
+    def get_model_contract_hash(self):
+        return md5_hash(
+            self.dict(
+                include=ModelContractConfig.__fields__.keys()
+                - CommonFieldsConfig.__fields__.keys(),
```

Comment:
Is there a reason for excluding `CommonFieldsConfig.__fields__.keys()` and not `exclude_fields_from_cache(self)`? It's just simpler?
```suggestion
                exclude=exclude_fields_from_cache(self),
```

Reply:
I want to remove everything in `ProjectConfig` too here.

Reply:
Oh right, because `ProjectConfig` in turn inherits from other classes, and we want to exclude those fields. I see. It was not obvious to me. Do you think we should add a comment justifying what we want here?

Reply:
I can add a small docstring!